packages/server-bridge/src/attachEngineIoBridgeToServer.js

```diff
@@ -68,63 +61,62 @@ export default function attachEngineIoBridgeToServer(originalConfig) {
+          instrumentation.done(e);
+          reportError(e, session, command);
+        });
     });
   }
 
   function onConnection(socket) {
     instrumentation.numOfConnectedClients(Date.now(), size(server.clients));
     log.info(`${stellarRequest.source} @StellarBridge: New Connection`);
-    const startTime = Date.now();
-
-    const requestUrl = get(socket, 'request.url');
-    const parsedUrl = url.parse(requestUrl, true);
-    const socketId = socket.id;
-    const sessionId = get(parsedUrl, 'query.x-sessionId') || socketId;
-
-    const initialSession = startSession({
-      ip: get(socket, 'request.connection.remoteAddress'),
-      sessionId,
-      socketId,
-      logPrefix: `${stellarRequest.source} @StellarBridge(${socketId}, ${sessionId})`,
-    });
 
-    socket.on('error', () => log.info(`${initialSession.logPrefix} Error`));
+    const socketSession = {
+      socketId: socket.id,
+      defaultSessionId: socket.id,
+      client: new WebsocketTransport(socket, stellarRequest.source, log, true),
+    };
+
+    const initialSession = startSession(socket.request, socketSession);
+
+    socket.on('error', () => log.info(`Error`, initialSession.logPrefix));
 
     const initialOnClose = () => onClose(initialSession);
     socket.on('close', initialOnClose);
 
-    callHandlersSerially(_newSessionHandlers, { source: stellarRequest.source, socket, session: initialSession })
-            .then((session) => {
-              sessions[socket.id] = session; // eslint-disable-line better-mutation/no-mutation
-              log.info(`${session.logPrefix} Connected`, pick(session, ['sessionId']));
-
-              socket.removeListener('close', initialOnClose);
-              socket.on('close', () => onClose(session));
-              socket.on('message', str => onMessage(str, session));
-
-              instrumentation.sessionStarted(Date.now() - startTime, session);
-              const hiMessage = {
-                messageType: 'connected',
-                message: 'connected to stellar bridge',
-                userId: session.authenticatedUserId,
-                sessionId: session.sessionId };
-              return socket.send(JSON.stringify(hiMessage));
-            })
-            .catch((e) => {
-              reportError(e, initialSession);
-              log.error(e, `${initialSession.logPrefix} Connection error`);
-              instrumentation.sessionFailed(Date.now() - startTime);
-              const errorMessage = { messageType: 'error', errorType: e.constructor.name, message: e.message, status: 401 };
-              socket.send(JSON.stringify(errorMessage));
-            });
+    callHandlersSerially({ request: socket.request, session: initialSession })
+      .then((session) => {
+        sessions[socket.id] = session; // eslint-disable-line better-mutation/no-mutation
+        log.info(`Connected`, session.logPrefix);
+
+        socket.removeListener('close', initialOnClose);
```

Comment:
Why do you setup close event to begin with?

Reply:
in case the close happens during session setup. i dont want to miss the event